backend/src/modules/installment/resolvers/undo-installment.ts

```
@@ -0,0 +1,32 @@
+import Bill from '../../../models/bill';
+import Installment from '../../../models/installment';
+
+type Args = {
+  id: string;
+};
+
+const undoInstallment = async (_parent: any, args: Args) => {
+  const { id } = args;
+
+  const installment = await Installment.findById(id);
+
+  if (!installment) throw new Error('The installment does not exist');
+
+  const bill = await Bill.findById(installment.bill);
+
+  if (!bill) throw new Error('Unexpected error');
+
+  if (installment.isPaid) {
+    installment.isPaid = false;
+
+    await installment.save();
```

We need to check if this succeeds before modifying the balance

---

Maybe just throw an error if it fails to interrupt the execution